src/txn/txn_ckpt.c

```diff
@@ -1440,8 +1451,10 @@ __checkpoint_tree(
 	 * If the checkpoint didn't complete successfully, make sure the
 	 * tree is marked dirty.
 	 */
-	if (ret != 0 && !btree->modified && was_modified)
+	if (ret != 0 && !btree->modified && was_modified) {
 		btree->modified = 1;
+		S2C(session)->modified = 1;
```

Comment:
Can we check that `conn->modified == 0` before setting it?  Otherwise, there could be lots of updates to that shared location here (this path happens a lot in some workloads).